python/ray/tune/ray_trial_executor.py

```diff
@@ -213,12 +213,13 @@ def _return_resources(self, resources):
         assert self._committed_resources.gpu >= 0
 
     def _update_avail_resources(self):
-        clients = ray.global_state.client_table()
         if ray.worker.global_worker.use_raylet:
             # TODO(rliaw): Remove once raylet flag is swapped
-            num_cpus = sum(cl['Resources']['CPU'] for cl in clients)
-            num_gpus = sum(cl['Resources'].get('GPU', 0) for cl in clients)
+            resources = ray.global_state.available_resources()
```

Comment:
Note that this is a somewhat expensive call since it waits for heartbeats from each node (e.g., could take 100s of milliseconds).